docs/core/porting/cppcli.md

```diff
@@ -0,0 +1,88 @@
+---
```

Comment:
should the filename be cpp-cli? usually individual words are separated by hyphen

Reply:
Yes, that makes sense. I'll update it to be cpp-cli.